addons/hr_holidays/models/hr_leave.py

```diff
@@ -711,8 +711,9 @@ def _check_validity(self):
                 continue
             employee = leave.employee_id
             date_from = leave.date_from.date()
-            leave_data = leave_type.get_allocation_data(employee, date=date_from)
-            if leave_data[employee][0][1]['excess_days']:
+            leave_data = leave_type.get_allocation_data(employee, date_from)
+            max_excess = leave_type.max_allowed_negative if leave_type.allows_negative else 0
+            if leave_data[employee][0][1]['total_virtual_excess'] > max_excess:
                 raise ValidationError(_('The allocation configuration does not allow you to take this leave.'))
```

Comment:
I feel this error is not really descriptive. This is the error that you encounter when you're trying to take a leave but not have enough allocated days to take it. Of course the configuration could allow you to take more, but with this message I'd automatically assume there is some configuration problem or so.